rclpy/test/test_executor.py

```diff
@@ -87,6 +88,27 @@ def timer_callback():
         finally:
             executor.shutdown()
 
+    def test_shutdown_executor_before_waiting_for_callbacks(self):
+        self.assertIsNotNone(self.node.handle)
+        for cls in [SingleThreadedExecutor, MultiThreadedExecutor]:
+            executor = cls(context=self.context)
+            executor.shutdown()
+            self.assertRaises(
+                ShutdownException,
+                executor.wait_for_ready_callbacks
+            )
+
+    def test_shutdown_exception_from_callback_generator(self):
+        self.assertIsNotNone(self.node.handle)
+        for cls in [SingleThreadedExecutor, MultiThreadedExecutor]:
+            executor = cls(context=self.context)
+            cb_generator = executor._wait_for_ready_callbacks()
+            executor.shutdown()
+            self.assertRaises(
+                ShutdownException,
+                lambda: next(cb_generator)
+            )
```

Comment:
@audrow nit^N: you can do:

```python
with self.assertRaises(ShutdownException):
    next(cb_generator)
```

to avoid that extra lambda.

Reply:
Good call.  I didn't know that you could use assert raises as a context.  Thanks.